### ISSUE TITLE
Recreating species association plots

### ISSUE THREAD

**Issue by JayCalvert** [guiblanchet/HMSC#11]
Hi

When I run the waterbirds example to try to recreate the species association plot for the influence of the previous year (Figure 7c in Ovaskainen et al, 2017), I do not get the same plot as the one presented in the paper. I have not edited the birds.R file in any way, nor have I changed anything in the data. 

For some reason I end up with a plot where all associations are large and positive, rather than the range presented in the paper. 

![image](https://user-images.githubusercontent.com/35904267/35488846-dc5ddcb0-0486-11e8-9f4f-5d788706af88.png)

Ultimately I want to produce a similar plot for my own data, however I get the similar all positive results for associations. Do you have any suggestions about what might be happening here?

Cheers

Jay

**Comment by guiblanchet** [comment]
Hi !

I have just ran the code from the birds.R file and got the expected results (i.e. The ones found in Figure 7c of the Ovaskainen et al. (2017) paper. 

Your result is confusing and slightly worrying. If you run the analysis again on your computer but making sure you ran `rm(list=ls())` before running the code, do you obtain the same odd results ?

**Comment by JayCalvert** [comment]
Hi, thanks for getting back to me so quickly.

I get those results every time, even when I have just started a new R session. The code I am using for the plot, from the birds.R file is:

`paramXReorg <- aperm(model$results$estimation$paramX, c(2, 1, 3))
Abeta <- paramXReorg[9:nrow(paramXReorg), , ]
beta1 <- apply(Abeta, c(1, 2), mean)
support <- pmax(apply(Abeta>0, c(1, 2), sum), 
                apply(Abeta<0, c(1, 2), sum)) / 
                dim(model$results$estimation$paramX)[3]
corrplot(support, method = "color", 
         col = colorRampPalette(c("blue", "white", "red"))(200))` 

Just out of interest, I changed the corrplot input to beta1 instead of support, and got a plot that more closely resembles the first one in figure 7c:

![image](https://user-images.githubusercontent.com/35904267/35522544-90df2062-0514-11e8-98b1-f4af7ef42221.png)

**Comment by JayCalvert** [comment]
Hi, thanks for getting back to me so quickly.

I get those results every time, even when I have just started a new R session. The code I am using for the plot, from the birds.R file is:

`paramXReorg <- aperm(model$results$estimation$paramX, c(2, 1, 3))
Abeta <- paramXReorg[9:nrow(paramXReorg), , ]
beta1 <- apply(Abeta, c(1, 2), mean)
support <- pmax(apply(Abeta>0, c(1, 2), sum), apply(Abeta<0, c(1, 2), sum)) /dim(model$results$estimation$paramX)[3]
corrplot(support, method = "color", col = colorRampPalette(c("blue", "white", "red"))(200))`

Just out of interest, I changed the corrplot input to beta1 instead of support, and got a plot that more closely resembles the first one in figure 7c:

![image](https://user-images.githubusercontent.com/35904267/35522795-30bf4c1a-0515-11e8-8516-8d74121ab79d.png)

**Comment by guiblanchet** [comment]
This clarifies things. 

I was looking (by mistake) at the species association matrices which all result in the figure as the one in the paper. You are right, `beta1` should be plotted, not `support`. After checking the code carefully, this is a mistake in the birds.R file. 

In addition, I recently found that in certain fringe situations, using the `aperm` function as was done in the birds.R file (and as you presented in your code above) can lead to some unexpected (and wrong result). As such, it is better to use the following lines of code to perform these analyses, which remove the use of the `aperm` function.
 
`Names <- dimnames(model$results$estimation$paramX)`
`interParamXNames <- which(Names[[2]] %in% Names[[1]])`

`paramXEst <- model$results$estimation$paramX`
`interParamX <- t(apply(paramXEst[, interParamXNames,], 1:2, mean))`

`corrplot(interParamX, method = "color", 
         col = colorRampPalette(c("blue", "white", "red"))(200))`

Note that I used different object names than the ones in the birds.R file... I just wrote this quickly and forgot to pay attention to the object names.
